sdk/metric/reader_test.go

```diff
@@ -216,3 +216,10 @@ func TestDefaultTemporalitySelector(t *testing.T) {
 		assert.Equal(t, metricdata.CumulativeTemporality, DefaultTemporalitySelector(ik))
 	}
 }
+
+func TestReadersAreHashable(t *testing.T) {
+	var _ map[Reader]struct{} = map[Reader]struct{}{
+		NewManualReader():                  {},
+		NewPeriodicReader(new(fnExporter)): {},
+	}
+}
```

Comment:
This is testing that the Readers are comparable.  It would fail to compile if ManualReader or PeriodicReader had incompatible fields.

Reply:
Why not include the compile time check in the code, [similar to other places](https://github.com/open-telemetry/opentelemetry-go/blob/1f5b159161e0ff54df4677284fa7d6718dc9186d/handler.go#L30-L33)?

Reply:
Because you can't create a variable of `comparable`, e.g. https://go.dev/play/p/ZxSl3MrV2Ei

But that did inspire a different change, we could make a type of `ComparableReader` and change the WithReader option to use that as its constraint.  That way if someone used a different Reader, via embedding, it would still need to be `comparable`.  I will make a separate PR for that.